tests/glsl/shader-subgroup-arithmetic.slang

```diff
@@ -0,0 +1,296 @@
+// not testing glsl or spirv due to crashing issue from some odd code emitting which looks right but fails
+// TEST:SIMPLE(filecheck=CHECK_GLSL):  -allow-glsl -stage compute -entry computeMain -target glsl
+// TEST:SIMPLE(filecheck=CHECK_SPV):  -allow-glsl -stage compute -entry computeMain -target spirv
+
+// not testing hlsl due to missing impl
+// TEST:SIMPLE(filecheck=CHECK_HLSL): -allow-glsl -stage compute -entry computeMain -target hlsl
+// not testing cuda due to missing impl
+// TEST:SIMPLE(filecheck=CHECK_CUDA): -allow-glsl -stage compute -entry computeMain -target cuda 
+// not testing cpp due to missing impl
+// TEST:SIMPLE(filecheck=CHECK_CPP):  -allow-glsl -stage compute -entry computeMain -target cpp
+
+// not testing glsl or spirv due to crashing issue from some odd code emitting which looks right but fails
+// TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl
+// TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl -emit-spirv-directly
+#version 430
+
+// #define TEST_when_equallity_and_uint_cast_on_templates_work
+
+precision highp float;
+precision highp int;
+
+//TEST_INPUT:ubuffer(data=[9 0 0 0 0], stride=4):name=inputBuffer
+buffer MyBlockName
+{
+    uint data[];
+} inputBuffer;
+
+#ifdef TEST_when_equallity_and_uint_cast_on_templates_work
+// TEST_INPUT:ubuffer(data=[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0  0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0  0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0  0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0  0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0], stride=4):out,name=outputBuffer
+#else
+// TEST_INPUT:ubuffer(data=[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0], stride=4):out,name=outputBuffer
+#endif
+
+buffer MyBlockName2
+{
+    uint data[];
+} outputBuffer;
+
+layout(local_size_x = 32) in;
+
+
+#ifdef TEST_when_equallity_and_uint_cast_on_templates_work 
+__generic<N : __BuiltinArithmeticType, M : __BuiltinIntegerType> 
+void testBasedOnType(uint group, N test, M test2) {
+    
+    uint index = 21 * group;
+    uint v[7] = uint[7](0, 0, 0, 0, 0, 0, 0); // done to issolate unimplemented functions
+
+    outputBuffer.data[index + 0] = subgroupAdd<N>(test) == 32;
+    outputBuffer.data[index + 1] = subgroupMul<N>(test) == 1;
+    outputBuffer.data[index + 2] = subgroupMin<N>(test) == test;
+    outputBuffer.data[index + 3] = subgroupMax<N>(test) == test;
+    
+    outputBuffer.data[index + 4] = subgroupAnd<M>(test2) == 0;
+    outputBuffer.data[index + 5] = subgroupOr<M>(test2) == 63;
+    outputBuffer.data[index + 6] = subgroupXor<M>(test2) == 0;
```

Comment:
These are not a correctly syntax for GLSL.
You should call them without the template argument N.
```
subgroupAdd(test) == 32;
```

Reply:
👍 the tests don't work regardless, but I did remove the template attribute specifier (for the next commit)